kong/clustering.lua

```diff
@@ -466,13 +503,12 @@ function _M.init_worker(conf)
     end, "dao:crud")
 
     kong.worker_events.register(function(data)
-      local res, err = declarative.export_config()
-      if not res then
-        ngx_log(ngx_ERR, "unable to export config from database: " .. err)
+      if push_config_semaphore:count() <= 0 then
+        push_config_semaphore:post()
```

Comment:
```suggestion
      if push_config_semaphore:count() <= 0 then
        -- the following line always executes immediately after the `if` check
        -- because `:count` will never yield, end result is that the semaphore
        -- count is guaranteed to not exceed 1
        push_config_semaphore:post()
```

I felt it is a piece of knowledge that could be easily lost over time so better to be more verbose.